sys/net/gnrc/routing/rpl/gnrc_rpl_dodag.c

```diff
@@ -358,6 +362,17 @@ static gnrc_rpl_parent_t *_gnrc_rpl_find_preferred_parent(gnrc_rpl_dodag_t *doda
     dodag->my_rank = dodag->instance->of->calc_rank(dodag, 0);
     if (dodag->my_rank != old_rank) {
         trickle_reset_timer(&dodag->trickle);
+
+#ifdef MODULE_NIMBLE_RPBLE
+        nimble_rpble_ctx_t ctx;
+        memset(&ctx, 0, sizeof(ctx));
+        ctx.inst_id = dodag->instance->id;
+        memcpy(ctx.dodag_id, &dodag->dodag_id, 16);
+        ctx.version = dodag->version;
+        ctx.rank = dodag->my_rank;
+        ctx.role = dodag->node_status;
+        nimble_rpble_update(&ctx);
+#endif
```

Comment:
since you have this very same block again in `gnrc_rpl.c`, better move it to a function that can then be defined to be a no-op when `nimble_rpble` is not used